pkg/forwarder/transaction_container.go

```diff
@@ -48,8 +50,14 @@ func buildTransactionContainer(
 
 	if optionalDomainFolderPath != "" && storageMaxSize > 0 {
 		serializer := NewTransactionsSerializer(domain, apiKeys)
-		storage, err = newTransactionsFileStorage(serializer, optionalDomainFolderPath, storageMaxSize, transactionsFileStorageTelemetry{})
+		var maxStorage *forwarderMaxStorage
+		storageMaxSize := config.Datadog.GetInt64("forwarder_storage_max_size_in_bytes")
+		diskRatio := config.Datadog.GetFloat64("forwarder_storage_max_disk_ratio")
 
+		maxStorage, err = newForwarderMaxStorage(optionalDomainFolderPath, filesystem.NewDisk(), storageMaxSize, diskRatio)
+		if err == nil {
+			storage, err = newTransactionsFileStorage(serializer, optionalDomainFolderPath, maxStorage, transactionsFileStorageTelemetry{})
+		}
 		// If the storage on disk cannot be used, log the error and continue.
 		// Returning `nil, err` would mean not using `TransactionContainer` and so not using `forwarder_retry_queue_payloads_max_size` config.
```

Comment:
I think you should still return an error and this should be the responsibility of the caller to decide what to do (and in this case, as your comment is saying) to continue despite the error.

Reply:
The function `buildTransactionContainer` is not a constructor (`newTransactionContainer` is the constructor). 
`buildTransactionContainer` instantiates `TransactionContainer` struct and its dependencies as the code was designed using dependencies injection. This code was previously  in `domain_forwarder.go` but as the code started to grow too much, I created this function.
This is a common pattern to create a function that instantiate object and its dependencies  when not using DI container and when working with mix code (dependency injection + no dependency injection ).

Note: As you said, if this logic was in `newTransactionContainer` it would have been be an issue. 

In addition, from an abstraction perspective, `buildTransactionContainer` creates a container to store transactions with a limited capacity. It's an implementation detail whether or not the disk is used.